lib/medtronic/processData.js

```diff
@@ -0,0 +1,378 @@
+    }
+  }
+  return 'unknown';
+};
+
+var filterHistory = function (types, log_records) {
+  var neededLogIds = [];
+  types.forEach(function (element) { neededLogIds.push(element.value); });
+  return log_records.filter(function (record) {
+    return neededLogIds.indexOf(record.type.value) >= 0;
+  });
+};
+
+var buildWizardRecords = function (records) {
+  var wizardRecords = filterHistory([RECORD_TYPES.BOLUS_WIZARD], records);
+  var postrecords = [];
+  wizardRecords.forEach(function(record) {
+    console.log('Wizard:', record.jsDate, common.bytes2hex(record.date), common.bytes2hex(record.body));
+    var carbInput = struct.extractByte(record.body,0);
+    var bgRaw = [struct.extractByte(record.head,1)];
+    bgRaw = bgRaw.concat(struct.extractByte(record.body,1) & 0x0f);
+    var bgInput = struct.extractShort(bgRaw,0);
+    var carbRatio = struct.extractByte(record.body,14) / 10.0;
+    var bgTarget = {};
+    bgTarget.low = struct.extractByte(record.body,5);
+    var isf = struct.extractByte(record.body,4);
+    bgTarget.high = struct.extractByte(record.body,3);
+    var recommended = {};
+    recommended.carb = struct.extractBEShort(record.body,7) / 40.0;
+    recommended.correction = struct.extractByte(record.body,6) / 40.0;
+    recommended.net = struct.extractBEShort(record.body,12) / 40.0;
+    var iob =  struct.extractBEShort(record.body,10) / 40.0;
+
+    var wizard = cfg.builder.makeWizard()
+      .with_recommended({
+        carb: recommended.carb,
+        correction: recommended.correction,
+        net: recommended.net
+      })
+      .with_bgInput(bgInput)
+      .with_carbInput(carbInput)
+      .with_insulinOnBoard(iob)
+      .with_insulinCarbRatio(carbRatio)
+      .with_insulinSensitivity(isf)
+      .with_bgTarget({
+        low: bgTarget.low,
+        high: bgTarget.high
+      })
+      .with_units('mg/dL'); //TODO: see if we can find units
+
+      wizard = wizard.with_deviceTime(sundial.formatDeviceTime(record.jsDate))
+          .set('index', record.index)
+          .set('jsDate', record.jsDate);
+
+      cfg.tzoUtil.fillInUTCInfo(wizard, record.jsDate);
+      postrecords.push(wizard);
+
+  });
+
+  return postrecords;
+};
+
+var buildBolusRecords = function (records) {
+  var bolusRecords = filterHistory([RECORD_TYPES.BOLUS], records);
+  var postrecords = [];
+
+  bolusRecords.forEach(function(bolus) {
+    console.log('Bolus:',bolus.jsDate, common.bytes2hex(bolus.head), common.bytes2hex(bolus.date));
+  });
+
+  for(var i=0; i < bolusRecords.length; i++) {
+    var record = bolusRecords[i];
+
+    var bolus;
+
+    if (larger) {
+      var duration = record.head[7] * 30 * sundial.MIN_TO_MSEC;
+      var amount = struct.extractBEShort(record.head,3)/40.0;
+      var programmed = struct.extractBEShort(record.head,1)/40.0;
+      var iob = struct.extractBEShort(record.head,5)/40.0;
+      if (duration > 0) {
+        //TODO: put iob in payload?
+
+        if((record.date[2] & 0xC0) == 0x80) {
```

Comment:
Seems like there should be some constants factored out here. What is `0xC0`? What is `0x80`? Not easy to read what the logic is as it stands...